packages/@aws-cdk/aws-apigatewayv2/lib/websocket/api.ts

```diff
@@ -0,0 +1,164 @@
+import { Construct } from 'constructs';
+import { CfnApi } from '../apigatewayv2.generated';
+import { IApi } from '../common/api';
+import { ApiBase } from '../common/base';
+import { DomainMappingOptions, IStage } from '../common/stage';
+import { WebSocketRouteIntegrationConfig, WebSocketIntegration } from './integration';
+import { WebSocketRoute, WebSocketRouteOptions } from './route';
+import { WebSocketStage } from './stage';
+
+/**
+ * Represents a WebSocket API
+ */
+export interface IWebSocketApi extends IApi {
+  /**
+   * Add a websocket integration
+   * @internal
+   */
+  _addIntegration(scope: Construct, config: WebSocketRouteIntegrationConfig): WebSocketIntegration
+}
+
+/**
+ * Props for WebSocket API
+ */
+export interface WebSocketApiProps {
+  /**
+   * Name for the WebSocket API resoruce
+   * @default - id of the WebSocketApi construct.
+   */
+  readonly apiName?: string;
+
+  /**
+   * The description of the API.
+   * @default - none
+   */
+  readonly description?: string;
+
+  /**
+   * The route selection expression for the API
+   * @default '$request.body.action'
+   */
+  readonly routeSelectionExpression?: string;
+
+  /**
+   * The name of the default stage with deployment
+   * @default - No default stage is created
+   */
+  readonly defaultStageName?: string;
+
```

Comment:
Why not follow the same pattern as HttpApi here - `defaultStage` and `createDefaultStage` properties? (sidenote: for the latter, I would prefer calling it `disableDefaultStage`)

Reply:
Same as the previous comment regarding default stage name not working in WebSocket APIs

Reply:
I don't think I follow this. What is the difference between websocket and http apis when it comes to default stages? Can you expand?

Comparing the documentation for [websocket api](https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-set-up-websocket-deployment.html) and [http api](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-stages.html), perhaps the concept of a default stage doesn't exist on the former? In such a case, should we just drop support for this.

Reply:
So when I tried out the similar flow of `$default` stage for websocket api, there were some problems with the functionality. The cfn deployment was successful but the actual functionality was not working. I think it has to do with the name of the stage (probably containing `$`). I could not find any documentation around the same and hence decided to change the api a bit accordingly.

I agree on dropping the support or maybe diving deep with the apigw team regarding the same. Thoughts?

Reply:
Let's drop support for now. This can always be added later by you or anyone else with a proper deep dive.